include/llbuild/BuildSystem/BuildValue.h

```diff
@@ -33,6 +33,35 @@ class raw_ostream;
 namespace llbuild {
 namespace buildsystem {
 
+class CommandSignature {
+public:
+  CommandSignature() = default;
```

Comment:
I think it would mean we would want both a CommandSignature, and a CommandSignatureBuilder — the signature would just be the hash digest, but the builder could have the full state vector and expose the combine functions.